tasks/image-segmentation-yolov8/src/validate.py

```diff
@@ -0,0 +1,213 @@
+
+            predClassMask = predClassSegmentation["segmentation"]
+            decodedClassMask = maskCoco.decode(predClassMask)
+            predictedMask = np.logical_or(predictedMask, decodedClassMask).astype(int)
+
+            iouClass = iouScoreClass(sample.load().annotation.extractSegmentationMask(ImageDatasetClasses([clazz])), decodedClassMask)
+            iouClasses.append(iouClass)
+            csvSampleValues.append(f"{iouClass * 100:.2f}")
+
+        if len(iouClasses) > 0:
+            sampleAcc = sum(iouClasses) / len(iouClasses) * 100
+            csvSampleValues.append(f"{sampleAcc:.2f}")
+
+            with csvSamplesPath.open("a") as file:
+                writer = csv.DictWriter(file, fieldnames = fieldNamesSamples)
+                writer.writerow(dict(zip(fieldNamesSamples, csvSampleValues)))
+
+            plotSegmentationImage(sample, origSeg, predictedMask, sampleAcc, artifactImagesPath)
+
+    for path in artifactImagesPath.iterdir():
+        if path.is_file():
+            if taskRun.createArtifact(path, f"images/{path.name}") is None:
+                logging.warning(f">> [Image Segmentation] Failed to upload image \"{path.name}\" with segmentation as artifact")
+            else:
+                logging.info(f">> [Image Segmentation] The segmentation image \"{path.name}\" has been uploaded as artifact")
+
+    if taskRun.createArtifact(csvSamplesPath, csvSamplesPath.name) is None:
+        logging.warning(">> [Image Segmentation] Failed to upload csv file with samples results as artifact")
+    else:
+        logging.info(">> [Image Segmentation] The csv file with samples results has been uploaded as artifact")
+
+def validate(taskRun: TaskRun[ImageDataset]) -> float:
+    datasetPath = folder_manager.createTempFolder("dataset_val")
+    trainDatasetPath, validDatasetPath = prepareDataset(taskRun.dataset, datasetPath, 1.0)
+
+    yamlFilePath = datasetPath / "config.yaml"
+    createYamlFile(datasetPath, trainDatasetPath, validDatasetPath, taskRun.dataset.classes, yamlFilePath)
+
+    ctxModel: Model = taskRun.parameters["trainedModel"]
+    ctxModel.download()
+    model = YOLO(ctxModel.path / "best.pt")
+
+    valResPath = folder_manager.createTempFolder("validation_results")
+
+    logging.info(">> [Image Segmentation] Validating the model")
+    model.val(
+        project = valResPath,
+        data = yamlFilePath,
+        batch = taskRun.parameters["batchSize"],
+        imgsz = taskRun.parameters["imageSize"],
+        save_json = True,
+    )
+
+    jsonFiles = valResPath.rglob("predictions.json")
+    for path in jsonFiles:
+        jsonPath = path
+
+    with jsonPath.open("r") as file:
+        predictData = list(json.load(file))
+
+    csvSamplesPath = folder_manager.createTempFolder("csv_samples") / "sample_results.csv"
+
+    processPrediction(taskRun, predictData, csvSamplesPath)
+
+    accuracy = createCsvDatasetResults(taskRun, csvSamplesPath)
```

Comment:
This should not return accuracy. `processPrediction` function should return full results of the validation. `createCsvDatasetResults` should take those results and then create `dataset_results.csv` file which contains the results on the dataset level and `sample_results.csv` file which contains the result on the sample level. Each of the samples should also have a separate file `{sample.id}.csv` which contains the results per label for that sample.